DllImportGenerator/DllImportGenerator/ManualTypeMarshallingHelper.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System.Linq;
```

Comment:
Let's keep our Linq statements to a minimum. I appreciate they are convenient but they are also a huge perf sink.

Reply:
I've tried to keep them pretty limited to just a little bit of member lookup. I might make some more helper methods to limit usage.